idseq_dag/util/s3.py

```diff
@@ -26,13 +29,20 @@
 S3MI_SEM = multiprocessing.Semaphore(MAX_CONCURRENT_S3MI_DOWNLOADS)
 IOSTREAM_UPLOADS = multiprocessing.Semaphore(MAX_CONCURRENT_UPLOAD_OPERATIONS)
 
+s3 = boto3.resource('s3')
```

Comment:
Is this safe to use from multiple threads and from multiple processes?

Reply:
There is some uncertainty about how safe it is to use the same boto3 client instance from multiple threads, so I'm going to make each subroutine spawn its own client.

Reply:
Thank you.